### PR TITLE
refactor: api templates improvements

### PR DESCRIPTION
**Pull request by danvincent11**

### CODE REVIEW

File: src/lib/core/domain/repository/social.repository.ts [hunk 1]
```diff
@@ -6,7 +7,7 @@ import type {
 } from '$lib/core/domain/models/social.dto'
 
 export interface ISocialRepository {
-	CreatePost(request: CreatePostRequest): Promise<CreatePostResponse>
-	GetPosts(): Promise<PostResponse[]>
-	GetCommentsByPostId(postId: number): Promise<CommentResponse[]>
+	CreatePost(request: CreatePostRequest): Promise<APIResponse<CreatePostResponse>>
+	GetAllPosts(): Promise<APIResponse<PostResponse[]>>
+	GetPostComments(postId: number): Promise<APIResponse<CommentResponse[]>>
```
Comment (kabaluyot) [inline src/lib/core/domain/repository/social.repository.ts]: add comments for interface methods (all)

File: src/lib/servicecontainer.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+import type { ILocalStorageRepository as LocalStorageRepositoryInterface } from '$lib/core/domain/repository/localstorage.repository'
+import { LocalStorageRepository } from '$lib/core/infrastructures/repository/localstorage.repository'
 
 /// dependency injection
 const $axios = axios.create(AxiosConfig) // base axios
 
-export const SocialRepository: ISocialRepository = new SocialRepositoryImpl($axios)
-export const LocalStorageRepository: ILocalStorageRepository = new LocalStorageRepositoryImpl()
+export const ISocialRepository: SocialRepositoryInterface = new SocialRepository($axios)
+export const ILocalStorageRepository: LocalStorageRepositoryInterface = new LocalStorageRepository()
```
Comment (kabaluyot) [inline src/lib/servicecontainer.ts]: sakto ngiyang sang left side boy (current changes), ngawn sang other projects ta ing outdated

File: src/lib/core/infrastructures/repository/social.repository.ts [hunk 3]
```diff
@@ -14,72 +15,65 @@ export class SocialRepository implements ISocialRepository {
...
 	}
 
 	/**
 	 * Get posts
 	 *
-	 * @return  {Promise<PostResponse[]>}
+	 * @return  {Promise<APIResponse><PostResponse>[]}
 	 */
-	public async GetPosts(): Promise<PostResponse[]> {
+	public async GetAllPosts(): Promise<APIResponse<PostResponse[]>> {
```
Comment (kabaluyot) [inline src/lib/core/infrastructures/repository/social.repository.ts]: sakto da ngawn `GetPosts`

Comment (kabaluyot) [reply src/lib/core/infrastructures/repository/social.repository.ts]: apply to all

File: src/routes/+page.svelte [hunk 4]
```diff
@@ -32,29 +36,28 @@
 			}
 		})
 
-		socialAPI.getPostsStore.subscribe((getPostsState) => {
-			if (getPostsState.state.SUCCESS || getPostsState.state.FAILED) {
-				console.log(getPostsState.response.message)
+		const getAllPostsSubscription = socialAPI.getAllPostsStore.subscribe((getAllPostsState) => {
```
Comment (kabaluyot) [inline src/routes/+page.svelte]: getPostsSubscription

**Review by kabaluyot — CHANGES_REQUESTED** [review]

File: src/lib/servicecontainer.ts (hunk 2 above)
Comment (danvincent11) [reply src/lib/servicecontainer.ts]: done

File: src/lib/core/infrastructures/repository/social.repository.ts (hunk 3 above)
Comment (danvincent11) [reply src/lib/core/infrastructures/repository/social.repository.ts]: done

File: src/routes/+page.svelte (hunk 4 above)
Comment (danvincent11) [reply src/routes/+page.svelte]: done

File: src/lib/core/domain/repository/social.repository.ts (hunk 1 above)
Comment (danvincent11) [reply src/lib/core/domain/repository/social.repository.ts]: done

File: src/lib/core/domain/repository/social.repository.ts [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import type { APIResponse } from '../models/dto'
```
Comment (kabaluyot) [inline src/lib/core/domain/repository/social.repository.ts]: use full path import

**Review by kabaluyot — CHANGES_REQUESTED** [review]

File: src/lib/core/domain/repository/social.repository.ts (hunk 5 above)
Comment (danvincent11) [reply src/lib/core/domain/repository/social.repository.ts]: done

**Review by kabaluyot — APPROVED** [review]
LGTM

**Pull request merged**